editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -5160,6 +5191,19 @@ void VisualShaderEditor::_param_unselected() {
 	_clear_preview_param();
 }
 
+void VisualShaderEditor::_panning_debounce_timer_timeout() {
+	String id_string = visual_shader->get_path();
+	const ResourceUID::ID uid = ResourceLoader::get_resource_uid(id_string);
+	if (uid != ResourceUID::INVALID_ID) {
+		id_string = ResourceUID::get_singleton()->id_to_text(uid);
+	}
+
+	const int type = get_current_shader_type();
+	vs_editor_cache->set_value("visual_shader", id_string + ":type" + itos(type) + ":offset", graph->get_scroll_offset() / EDSCALE);
```

Comment:
The key strings could also be handled by a method. It's more reliable than manually constructing it every time.